Libraries/Boards/MAX32690/EvKit_V1/Source/board.c

```diff
@@ -131,6 +149,66 @@ int Console_PrepForSleep(void)
     return MXC_UART_ReadyForSleep(ConsoleUart);
 }
 
+/******************************************************************************/
+void TFT_SPI_Init(void)
+{
+    // Configure GPIOs
+    MXC_GPIO_Config(&tft_mosi);
+    MXC_GPIO_Config(&tft_clk);
+    MXC_GPIO_Config(&tft_cs);
+
+    // Set chip select and clock lines high (idle state)
+    MXC_GPIO_OutSet(tft_cs.port, tft_cs.mask);
+    MXC_GPIO_OutSet(tft_clk.port, tft_clk.mask);
+}
+
+/******************************************************************************/
+void TFT_SPI_Write(uint8_t *datain, uint32_t count, bool data)
+{
+    uint8_t tx_byte;
+
+    if (datain == NULL) {
+        return;
+    }
+
+    MXC_GPIO_OutClr(tft_cs.port, tft_cs.mask); // Assert chip select (active low)
+    MXC_Delay(10);
+
+    for (int i = 0; i < count; i++) {
+        tx_byte = datain[i];
+
+        for (int j = 0; j < 9; j++) { // 9 Bits per character
+            tft_clk.port->out_clr = tft_clk.mask; //Clk low
+
+            if (j == 0) { //Send D/CX bit before MSB
+                if (data) {
+                    tft_mosi.port->out_set = tft_mosi.mask;
+                } else {
+                    tft_mosi.port->out_clr = tft_mosi.mask;
+                }
+            } else {
+                if (tx_byte & 0x80) { //Shift out data MSB first
+                    tft_mosi.port->out_set = tft_mosi.mask;
+                } else {
+                    tft_mosi.port->out_clr = tft_mosi.mask;
+                }
+                tx_byte = tx_byte << 1;
+            }
+
+            for (int k = 0; k < 2; k++) {}
```

Comment:
These delay loops may not work if the compiler decides to be "smart" and optimize them away.  Does MXC_Delay not have good enough resolution?  A set of "__NOP();" instructions may be better, but then you're subject to whatever the system clock is running at.

Reply:
Had a hallway conversation with Jacob about this.  MXC_Delay doesn't offer a short enough delay due to the overhead it has of converting the passed in time to system tick values.  Removing these delays altogether also works, but results in a non-50% duty cycle.  Since this file is board specific anyway, we decided to go ahead and leave these SW delay loops as-is.